src/operators/join.rs

```diff
@@ -451,28 +488,20 @@ impl<G, T1> JoinCore<G, T1::Key, T1::Val, T1::R> for Arranged<G,T1>
                 if trace2.is_some() && input1.frontier().is_empty() { trace2 = None; }
```

Comment:
up to this point we did everything for {input,trace}1, then {input,trace}2 - this seems like the first instance where we do something for trace2 before trace1 - is the ordering important here?

Reply:
I think the ordering is not important, and it was original conceived as "handle `input1` being complete" but the logic hoists references to `trace2` up around it. I'll comment it up explaining the the order doesn't matter.